### PR TITLE
Add current and desired state for ebsvolume resource

### PR DESCRIPTION
**Pull request by rossf7**
Towards giantswarm/giantswarm#2544

Implements current and desired state. Deletion logic will be in a follow up PR.

### CODE REVIEW

File: service/awsconfig/v5/resource/ebsvolume/current.go [hunk 1]
```diff
@@ -2,9 +2,72 @@ package ebsvolume
...
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/giantswarm/apiextensions/pkg/apis/provider/v1alpha1"
+	"github.com/giantswarm/microerror"
+
+	"github.com/giantswarm/aws-operator/service/awsconfig/v5/key"
+)
+
+const (
+	cloudProviderClusterTagValue        = "owned"
+	cloudProviderPersistentVolumeTagKey = "kubernetes.io/created-for/pv/name"
```
Comment (rossf7) [inline service/awsconfig/v5/resource/ebsvolume/current.go]: The EBS volume is only deleted if it has the cluster tag and a PV name tag.

File: service/awsconfig/v5/resource/ebsvolume/current.go [hunk 2]
```diff
@@ -2,9 +2,72 @@ package ebsvolume
...
+
+	clusterTag := key.ClusterCloudProviderTag(customObject)
+	filters := []*ec2.Filter{
+		{
+			Name: aws.String(fmt.Sprintf("tag:%s", clusterTag)),
+			Values: []*string{
+				aws.String(cloudProviderClusterTagValue),
+			},
+		},
+	}
```
Comment (rossf7) [inline service/awsconfig/v5/resource/ebsvolume/current.go]: Here we filter to get only volumes with the cluster cloud provider tag.

File: service/awsconfig/v5/resource/ebsvolume/mock_test.go [hunk 3]
```diff
@@ -0,0 +1,46 @@
+package ebsvolume
+
+import (
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/giantswarm/apiextensions/pkg/apis/provider/v1alpha1"
+	"github.com/giantswarm/aws-operator/service/awsconfig/v5/key"
```
Comment (xh3b4sd) [inline service/awsconfig/v5/resource/ebsvolume/mock_test.go]: Lets please separate the import for the same project from third parties.

Comment (rossf7) [reply service/awsconfig/v5/resource/ebsvolume/mock_test.go]: Ack, missed that. Done.

**Pull request merged**